03week/towersOfHanoi.js

```diff
@@ -19,24 +50,56 @@ function printStacks() {
   console.log("c: " + stacks.c);
 }
 
-function movePiece() {
-  // Your code here
-
+function movePiece(startStack, endStack) {
+  stacks[endStack].push(stacks[startStack].pop());
+  //moves the top stone from the start stack to the top of the end stack
 }
 
-function isLegal() {
-  // Your code here
-
+function isLegal(startStack, endStack) {
+  const startArr = stacks[startStack];
+  const endArr = stacks[endStack];
+  if (!startArr[0])return false
+  //you can't move something from nothing
+  if (!endArr[0])return true
+  //if you have something to move you can always move it to an empty stack
+  if (startArr[startArr.length -1] < endArr[endArr.length -1]) return true;
+  /*if the top stone in the starting stack is smaller than the top stone of the
+  ending stack, you are in the clear. */
+  return false;
 }
 
 function checkForWin() {
-  // Your code here
+  return Boolean(stacks.b.length === 4)
+  //if stackb has 4 stones, you win!
+}
 
+function resetStacks() {
+  stacks.a = [4, 3, 2, 1]; //full
+  stacks.b = [];// empty
+  stacks.c = []; //empty
 }
 
 function towersOfHanoi(startStack, endStack) {
-  // Your code here
-
+  const legalStacks = ['a','b','c'];
```

Comment:
good but you could have organized this a little cleaner so the code would be more performant. 
` if (legalStacks.indexOf(startStack) === -1) {
      if (startStack !== endStack) {.... }
}